server/client.go

```diff
@@ -1326,17 +1326,29 @@ func (c *client) closeConnection() {
 	var (
 		routeClosed   bool
 		retryImplicit bool
+		connectURLs   []string
 	)
 	if c.route != nil {
 		routeClosed = c.route.closed
 		if !routeClosed {
 			retryImplicit = c.route.retry
 		}
+		connectURLs = c.route.connectURLs
```

Comment:
Can this change out from underneath of us since we release the lock on next line?

Reply:
No, the c.route.connectURLs is an array that we get when getting the INFO from that route. It won't change. If the route disconnects and reconnects it is a brand new c (client).